micro-rdk/src/esp32/analog.rs

```diff
@@ -42,4 +42,17 @@ impl<'a, const A: u32, T: ADCPin> AnalogReader<u16> for Esp32AnalogReader<'a, A,
     fn name(&self) -> String {
         self.inner_name()
     }
+    fn resolution(&self) -> crate::common::analog::AnalogResolution {
+        // TODO: In esp32/board.rs we have selected the ADC_ATTEN_DB_11 setting
+        // for our ADC drivers, resulting in the values below (see docs on ESP-IDF for more info).
+        // If and when we make this configurable, this function's implementation will have to change
+        // as well
+
+        // all units are in mV
+        AnalogResolution {
+            min_range: 150.0,
+            max_range: 2450.0,
```

Comment:
true for the linear range for the ADC can actually measure higher than 2.45mv might make more sense to use `esp_adc_cal_raw_to_voltage` to feed min max and resolution (0-4095) into this struct.

Reply:
it's far from perfect since adc curve are not linear for error will accumulate especially outside the 150-2450 range

Reply:
sorry, I'm a little unclear here. That function seems to directly compute voltage, so do you mean I should replace the raw value with the value in mV? If so does it actually make sense to even return non-trivial values for the resolution?

Reply:
The range in this struct is the suggested voltage range for esp32 for accurate voltage measurement where raw to volatge is mostly linear. That doesn't mean ADC is not able to measure voltage outside of this range. I don't know if the purpose of this api is to offload the conversion to SDKs but if it is the case then we should calcultage mon/max voltage by calling ` esp_adc_cal_raw_to_voltage(0,&chars)` for min and ` esp_adc_cal_raw_to_voltage(4095,&chars)` for min/max (can use the rust bindings for that too) and assume it's linear for the range (which is not) Would need to confirm the values are not too insane

Ideally we should return mV instead of raw

Reply:
I agree with the last sentence, I wonder if we can just provide the voltage and abide by some standard for the existing resolution values?

Reply:
:ok_hand:

Reply:
if the returned value goes above 2450 what's happening on the SDK side?